scripts/setup.py

```diff
@@ -563,11 +283,33 @@ def setup_nss_slurm():
     run(r"sed -i 's/\(^\(passwd\|group\):\s\+\)/\1slurm /g' /etc/nsswitch.conf")
 
 
+def call_setup_network_storage(log):
+    try:
+        subprocess.run(
+            "/slurm/scripts/network_storage_wrappers/run_setup_network_storage.sh"
+        )
+        log.info("network storage mounted successfully")
+    except Exception as e:
+        log.error(e)
+
+
+def call_setup_nfs_exports(log):
+    try:
+        subprocess.run(
+            "/slurm/scripts/network_storage_wrappers/run_setup_nfs_exports.sh"
+        )
+        log.info("nfs exported successfully")
+    except Exception as e:
+        log.error(e)
+
+
 def setup_sudoers():
     content = """
 # Allow SlurmUser to manage the slurm daemons
 slurm ALL= NOPASSWD: /usr/bin/systemctl restart slurmd.service
 slurm ALL= NOPASSWD: /usr/bin/systemctl restart slurmctld.service
+slurm ALL=(ALL) NOPASSWD: /slurm/scripts/network_storage_wrappers/run_setup_network_storage.sh
```

Comment:
Strictly speaking, the second ALL allows slurm to run this script as any user, not just root. I would probably replace every instance of "ALL=" with "ALL=(root:root)" for explicitness since the implicit assumption here is that no one has modified the runas_default value from the default of "root"

>      A Runas_Spec determines the user and/or the group that a command may be run as.  A fully-specified Runas_Spec consists of two Runas_Lists (as defined
>     above) separated by a colon (‘:’) and enclosed in a set of parentheses.  The first Runas_List indicates which users the command may be run as via the -u
>     option.  The second defines a list of groups that may be specified via the -g option (in addition to any of the target user's groups).  If both
>     Runas_Lists are specified, the command may be run with any combination of users and groups listed in their respective Runas_Lists. If only the first is
>     specified, the command may be run as any user in the list and, optionally, with any group the target user belongs to.  If the first Runas_List is empty
>     but the second is specified, the command may be run as the invoking user with the group set to any listed in the Runas_List.  If both Runas_Lists are
>     empty, the command may only be run as the invoking user and the group, if specified, must be one that the invoking user is a member of.  If no Runas_Spec
>     is specified, the command may only be run as the runas_default user (root by default) and the group, if specified, must be one that the runas_default user
>     is a member of.

Reply:
Thanks for pointing this out. Updated.